redditInterface.py

```diff
@@ -0,0 +1,115 @@
+from fastapi import FastAPI
+from fastapi import HTTPException
+
+app = FastAPI()
+
+# Dataframe and return data
+redditFrame = []
+nFrames = 0
+
+
+# API Calls
+@app.get("/")
+def read_root():
+    raise HTTPException(status_code=404, detail="Invalid request url")
+
+
+@app.post("/subreddits")
+def add_subreddit(subreddit: str):
+    global redditFrame
+    global nFrames
+    subDict = {}
+    id = nFrames
+    nFrames += 1
+
+    subDict['id'] = id
+    subDict['Title'] = subreddit
+    subDict['Posts'] = []
+    subDict['number'] = 0
+    redditFrame.append(subDict)
+
+    return subDict
+
+
+@app.put("/subreddits/{subreddit_id}")
+def add_post(subreddit_id: int, post: str, post_metadata: str):
+    global redditFrame
+    postDict = {}
+    index = search_by_subreddit_id(subreddit_id)
+    subreddit = redditFrame[index]
+
+    if subreddit:
+        postDict['id'] = subreddit['number']
+        postDict['Title'] = post
+        postDict['Data'] = post_metadata
+        redditFrame[index]['Posts'].append(postDict)
+        redditFrame[index]['number'] += 1
+        return postDict
+    else:
+        raise HTTPException(status_code=404, detail="Subreddit not found")
+
+
+@app.get("/subreddits/{subreddit_id}")
+def read_song(subreddit_id: int):
```

Comment:
Is there a reason why these names all have `song` in them?

Reply:
That was a typo, I was testing it on other types of data before